pandas/core/groupby/groupby.py

```diff
@@ -2127,7 +2127,8 @@ def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None,
                                                      limit=limit, freq=freq,
                                                      axis=axis))
         filled = getattr(self, fill_method)(limit=limit)
-        filled = filled.drop(self.grouper.names, axis=1)
+        filled = filled.drop(filled.columns.intersection(self.grouper.names),
```

Comment:
What does this do that isn't already handled by the changes in generic?

Reply:
The code for `pct_change` was written after bug #21521, and assumes the buggy behaviour, causing various tests to fail when you apply the rest of the PR.

Specifically, line 2129 performs e.g. a groupby ffill, and then line 2130 drops all group labels from the result. As we're no longer returning all group labels, this would raise. The change fixes line 2130 to only drop those group labels which are present in the output.

Reply:
Makes sense. Thanks for clarifying